iree/compiler/Dialect/VMLA/IR/VMLABase.td

```diff
@@ -160,19 +160,25 @@ class VMLA_ElementTypeOp<string mnemonic, list<OpTrait> traits = []> :
   }];
 }
 
-class VMLA_UnaryOp<string mnemonic, Attr typeAttr, list<OpTrait> traits = []> :
-    VMLA_ElementTypeOp<mnemonic, traits> {
+class VMLA_UnaryOp<string mnemonic, Attr typeAttr, list<OpTrait> traits = [
+  AllTypesMatch<["src", "dst"]>
+  ]> : VMLA_ElementTypeOp<mnemonic, traits> {
   let arguments = (ins
     VMLA_Buffer:$src,
     VMLA_Buffer:$dst,
     typeAttr:$element_type,
     // TODO(benvanik): remove once unsigned types are in MLIR.
     UnitAttr:$forceUnsigned
   );
+
+  let assemblyFormat = [{
+    `(`$src`,` $dst`)` attr-dict`:` type($dst) $element_type
+  }];
```

Comment:
Since this is short, there's no need to put it in a codeblock (`[{...}]`). Quotes is simpler IMO. Combined with my other suggestion, this would look like:

```suggestion
  let assemblyFormat = "`(`$src`,` $dst`)` attr-dict`:` $element_type";
```